Makefile

```diff
@@ -1,41 +1,80 @@
 SHELL := /bin/bash
+include functions.mk
+
+# Name of the exporter
+EXPORTER_NAME := ebs-iops-reporter
+# valid: deployment or daemonset
+# currently unused
+EXPORTER_TYPE := deployment
 
 # All of the source files which compose the monitor. 
 # Important note: No directory structure will be maintained
 SOURCEFILES ?= monitor/main.py monitor/start.sh
 
+# What to prefix the name of resources with?
+NAME_PREFIX ?= sre-
+SOURCE_CONFIGMAP_SUFFIX ?= -code
+CREDENITALS_SUFFIX ?= -aws-credentials
+
+MAIN_IMAGE_URI ?= quay.io/jupierce/openshift-python-monitoring
 IMAGE_VERSION ?= stable
+INIT_IMAGE_URI ?= quay.io/lseelye/yq-kubectl
 INIT_IMAGE_VERSION ?= 1903.0.0
 
-RESOURCELIST := servicemonitor/ebs-iops-reporter service/ebs-iops-reporter \
-	deployment/ebs-iops-reporter secret/ebs-iops-reporter-credentials-volume \
-	configmap/ebs-iops-reporter-code rolebinding/sre-ebs-iops-reporter \
-	serviceaccount/sre-ebs-iops-reporter clusterrole/sre-allow-read-cluster-setup \
-	rolebinding/sre-ebs-iops-reporter-read-cluster-setup CredentialsRequest/ebs-iops-reporter-aws-credentials \
-	secrets/ebs-iops-reporter-aws-credentials
+# Generate variables
+
+MAIN_IMAGE ?= $(MAIN_IMAGE_URI):$(IMAGE_VERSION)
+INIT_IMAGE ?= $(INIT_IMAGE_URI):$(INIT_IMAGE_VERSION)
+
+PREFIXED_NAME ?= $(NAME_PREFIX)$(EXPORTER_NAME)
+
+AWS_CREDENTIALS_SECRET_NAME ?= $(PREFIXED_NAME)$(CREDENITALS_SUFFIX)
+SOURCE_CONFIGMAP_NAME ?= $(PREFIXED_NAME)$(SOURCE_CONFIGMAP_SUFFIX)
+SERVICEACCOUNT_NAME ?= $(PREFIXED_NAME)
+
+RESOURCELIST := servicemonitor/$(PREFIXED_NAME) service/$(PREFIXED_NAME) \
+	deploymentconfig/$(PREFIXED_NAME) secret/$(AWS_CREDENTIALS_SECRET_NAME) \
+	configmap/$(SOURCE_CONFIGMAP_NAME) rolebinding/$(PREFIXED_NAME) \
+	serviceaccount/$(SERVICEACCOUNT_NAME) clusterrole/sre-allow-read-cluster-setup \
+	rolebinding/sre-ebs-iops-reporter-read-cluster-setup CredentialsRequest/$(AWS_CREDENTIALS_SECRET_NAME)
+
 
-all: deploy/025_sourcecode.yaml deploy/040_deployment.yaml
+all: deploy/010_serviceaccount-rolebinding.yaml deploy/020-awscredentials-request.yaml deploy/025_sourcecode.yaml deploy/040_deployment.yaml deploy/050_service.yaml deploy/060_servicemonitor.yaml
+
+deploy/020-awscredentials-request.yaml: resources/020-awscredentials-request.yaml.tmpl
+	@$(call generate_file,020-awscredentials-request)
```

Comment:
Need to include exporter name in generated file, so another argument is required to `generate_file`.